x/sophon/keeper/keeper.go

```diff
@@ -146,14 +146,33 @@ func (k Keeper) GetSophonUserByPubKey(ctx sdk.Context, sophonPubKey []byte, user
 	return k.GetSophonUser(ctx, sophonInfo.Id, userID)
 }
 
-func (k Keeper) HasSophonTransfer(ctx sdk.Context, sophonID uint64, address []byte) (bool, error) {
-	return k.sophonTransfer.Has(ctx, collections.Join(sophonID, address))
+func (k Keeper) GetSophonTransfer(ctx sdk.Context, sophonID uint64) ([]byte, error) {
+	iterator, err := k.sophonTransfer.Iterate(ctx, collections.NewPrefixedPairRange[uint64, []byte](sophonID))
+	if err != nil {
+		return nil, err
+	}
+
+	// There should only be one transfer per sophon so collecting all keys is fine
+	keys, err := iterator.Keys()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(keys) == 0 {
+		return nil, collections.ErrNotFound
+	}
+
+	return keys[0].K2(), nil
 }
 
 func (k Keeper) SetSophonTransfer(ctx sdk.Context, sophonID uint64, address []byte) error {
 	return k.sophonTransfer.Set(ctx, collections.Join(sophonID, address))
 }
 
+func (k Keeper) DeleteSophonTransfer(ctx sdk.Context, sophonID uint64) error {
+	return k.sophonTransfer.Clear(ctx, collections.NewPrefixedPairRange[uint64, []byte](sophonID))
```

Comment:
Wouldn't it be better to have `sophonTransfer` as a map type from sophon ID to address so that we can just use `Remove` here?

Reply:
Yes that makes much more sense :)